src/Catalyst.Core.Modules.Ledger/Web3EthApi.cs

```diff
@@ -0,0 +1,46 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using Catalyst.Abstractions.Kvm;
+using Catalyst.Abstractions.Ledger;
+using Nethermind.Store;
+
+namespace Catalyst.Core.Modules.Ledger 
+{
+    public class Web3EthApi : IWeb3EthApi
+    {
+        public Web3EthApi(IDeltaExecutor deltaExecutor, IStateReader stateReader, IStateRootResolver stateRootResolver, IDeltaResolver deltaResolver)
+        {
+            DeltaExecutor = deltaExecutor ?? throw new ArgumentNullException(nameof(deltaExecutor));
+            StateReader = stateReader ?? throw new ArgumentNullException(nameof(stateReader));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)